plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -142,21 +151,27 @@ public static List<DeltaLakeColumnHandle> extractPartitionColumns(List<DeltaLake
     }
 
     public static String serializeSchemaAsJson(
+            ColumnMappingMode columnMappingMode,
             List<DeltaLakeColumnHandle> columns,
+            AtomicInteger newFieldId,
```

Comment:
```suggestion
OptionalInt newFieldId,
```

Reply:
The logic depends on `AtomicInteger` to propagate the max id. I won't address this comment.